spoon/template/compiler.php

```diff
@@ -775,28 +775,28 @@ protected function parseOptions($content)
 						$replace[] = '<?php
 						if(
 							(is_object(' . $baseVariable . ') && ' . $objectVariable . ' && ' . $objectVariable . ' != \'\' && ' . $objectVariable . ' !== false)
-							|| (is_array(' . $baseVariable . ') && isset(' . $variable . ') && count(' . $variable . ') != 0 && ' . $variable . ' != \'\' && ' . $variable . ' !== false))
+							|| (is_array(' . $baseVariable . ') && isset(' . $variable . ') && empty(' . $variable . ') === false && ' . $variable . ' != \'\' && ' . $variable . ' !== false))
 						{
 							?>';
 						$replace[] = '<?php } ?>';
 
 						// inverse option
 						$replace[] = '<?php if(
 							(is_object(' . $baseVariable . ') && (!' . $objectVariable . ' || ' . $objectVariable . ' == \'\' || ' . $objectVariable . ' === false))
-							|| (is_array(' . $baseVariable . ') && (!isset(' . $variable . ') || count(' . $variable . ') == 0 || ' . $variable . ' == \'\' || ' . $variable . ' === false))): ?>';
+							|| (is_array(' . $baseVariable . ') && (!isset(' . $variable . ') || empty(' . $variable . ') === true || ' . $variable . ' == \'\' || ' . $variable . ' === false))): ?>';
 						$replace[] = '<?php endif; ?>';
 					}
 					else
 					{
 						// replace with
 						$replace[] = '<?php
-						if(isset(' . $variable . ') && count(' . $variable . ') != 0 && ' . $variable . ' != \'\' && ' . $variable . ' !== false)
```

Comment:
maybe just add an is_array here as well?

Reply:
and by checking it is an array we also don't need to check if it is an empty string or false anymore I think

Reply:
I don't know by heart, but is the variable always going to be an array here? If not it would be strange to add this rule here, no?

Reply:
I don't know :) it sometimes is hard to know what ends up where in legacy projects